### DISCUSSION TITLE
Security concerns on URL Path metadata and Proxy disclosure HTTP Method

### DISCUSSION THREAD

**Discussion by Zeoulz** [minio/minio#20772] [Q&A] [answered]
Hi, i have a question based on security related. This information i got it when running OwaspZap.

1. What are the purpose of this path? example: https://example.com/latest/meta
2. Does Minio Community edition have configuration to disable specific HTTP method? Example like TRACE or OPTIONS?

**Comment by marktheunissen** [comment]
Hi @Zeoulz I think you have some generic information that is not relevant to MinIO

> What are the purpose of this path? example: https://example.com/latest/meta

That is not a MinIO path. If you believe it is, please show us what you're doing? 

> Does Minio Community edition have configuration to disable specific HTTP method? Example like TRACE or OPTIONS?

We are S3 compatible, so no we don't even handle TRACE. For OPTIONS we support CORS the same as S3 does.

Please be much more specific and provide evidence of your claims before raising "security" issues.

**Comment by Zeoulz** [comment]
Hi @marktheunissen 

Currently i have MinIO setup running in my internal server with MinIO console URL https://example.com:9001/  and MinIO API URL https://example.com:9000/.

When we do OwaspZap scan only required to put URL of the MinIO. After scan, report provide list of vulnerabilities which one of it is **Cloud Metadata Potentially Exposed** and **Proxy Disclosure**

![Proxy-Disclosure](https://github.com/user-attachments/assets/2268d551-4fc7-4322-9cd0-620928613b37)
![Cloud Metadata Potentially Exposed](https://github.com/user-attachments/assets/e4766f24-eabb-49b0-9f96-fe43704aa9d4)

FYI: Installation using binary method based on below guide
https://min.io/docs/minio/linux/operations/install-deploy-manage/deploy-minio-single-node-single-drive.html

**Reply by klauspost** [reply]
No idea what your security software is on about. Maybe they cannot imagine that S3 headers can come from sources other than AWS. I suggest reaching out to them.

I see nothing actionable here.

**Reply by marktheunissen** [reply, marked as answer]
@Zeoulz Neither of these is relevant, you have a "proxy server detected" which is incorrect. You've connected directly to MinIO. The only "proxy", if you could call it that, is the console making requests to the MinIO admin API which is by design and not knowledge leak. This is not  a security issue.

The second report is also false alarm, returning 200 OK for path /latest/meta-data is not "leaking cloud metadata".

**Comment by Zeoulz** [comment]
@marktheunissen thank you for the clarification.
